python/ray/tests/test_state_api.py

```diff
@@ -3080,18 +3078,14 @@ def ready(self):
 
     # Make sure when the --detail option is specified, the default formatting
     # is yaml. If the format is not yaml, the below line will raise an yaml exception.
-    print(
-        yaml.load(
-            result.output,
-            Loader=yaml.FullLoader,
-        )
-    )
+    # Retrieve yaml content from result output
+    print(yaml.safe_load(result.output.split("---")[1].split("...")[0]))
```

Comment:
I am a bit not sure the added benefit of using safe_load here versus the un-readability of the code with string manipulation. 

This is part of the test code and I am not sure if "safe" loading is that necearry? Could you maybe help me understand the risk of using unsafe loading here a bit more?

Reply:
I'm with you that this is test code and it doesn't take an API/external input.

We have internal systems which track any insecure methods and raise flags. This method is primarily prone for remote-execution and is encouraged to move away from it, instead use safe_load.

Regarding, result output parsing I consistently get ACTOR WARNINGs (with special characters) in the output and the safe_loading fails. To resolve the issue, I retrieve the yaml content from result based on start/end.

I run the test at two different environments (Mac/Linux) and I see same WARNINGs  (with special characters) in result output. Let me know your thoughts.